dts/bindings/led/led-controller.yaml

```diff
@@ -0,0 +1,43 @@
+# Copyright (c) 2020 Seagate Technology LLC
+# SPDX-License-Identifier: Apache-2.0
+
+# Common fields for LED controllers and child LEDs
+
+properties:
+    label:
+      required: true
+    num-leds:
+      type: int
+      required: false
+      description: Number of LEDs available.
+
+child-binding:
+    description: LED child node
+    properties:
+        label:
+          required: false
+          type: string
+          description: Human readable string describing the LED
+        index:
+          required: false
+          type: int
+          description: |
+            Index of the LED on a controller. It can be used by drivers or
+            applications to map a logical LED to its real position on the
+            controller. For example, this allows to handle boards where the
+            LEDs in an array/strip are not wired following the LED order of
+            the controller.
+        color-mapping:
+          type: array
```

Comment:
Not a uint8-array? That seems a bit odd compared with this signature:

```
led_set_color(struct device *dev, uint32_t led, uint8_t *color);
```

Using uint8-array would save space in the led_info arrays.

Reply:
Ah, never mind, the color value is not the same as a color channel value. But it still looks like color-mapping could fit in a uint8_t array in C, which would still save that space.

Reply:
Yes, it must fit an uint8_t array in C. I'll fix that. But I think it is better to keep the "array" property type here. This allows ot use the define from include/dt-bindings/led/led.h as they are in both C code and DT. With the uint8-array type, two digits (hexadecimal) are need to set the array values in DT.

Reply:
Sounds reasonable